README.md

```diff
@@ -1,6 +1,15 @@
-# External Data Provider
+# Kubernetes API Provider
 
-A template repository for building external data providers for Gatekeeper.
+The Kubernetes API provider is used for querying Kubernetes resources. This can be a 
+good alternative to caching, as the cache can grow larger as the cluster becomes
+busier with many resources.
+
+This provider offers an example of how to query Kubernetes objects where a
+cluster-scoped view is required to execute a policy. The example here demonstrates
+how uniqueness of the ingress host name is evaluated using this provider.
+
+Please refer to the [Quick Start](#quick-start) to set up your local environment and see a demo of
+this provider in action.
 
```

Comment:
please specify the steps in the `local-setup.sh` in the readme so users can learn, and then you can provide the script for automation

example: https://github.com/sozercan/trivy-provider